src/Adapter/PDF/OrderReturnPdfGenerator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\PDF;
+
+use Context;
+use PDF;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\Repository\OrderReturnRepository;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\OrderReturnSettings;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\ValueObject\OrderReturnId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\PDF\PDFGeneratorInterface;
+use RuntimeException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Validate;
+
+/**
+ * Generates invoice for given order.
+ */
+final class OrderReturnPdfGenerator implements PDFGeneratorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var OrderReturnRepository
+     */
+    private $orderReturnRepository;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param OrderReturnRepository $orderReturnRepository
+     */
+    public function __construct(TranslatorInterface $translator, OrderReturnRepository $orderReturnRepository)
```

Comment:
The injected repository should be behind an interface, so we can easily replace it.

Reply:
I am not sure about repository interface because main function get for both OrderReturnRepository and OrderRepository returns legacy objectModel (Order and OrderReturn). When we are replacing repositories we will probably refactor object model as well so such interfaces would have to be refactored too.